mlflow/server/auth/__init__.py

```diff
@@ -71,15 +373,36 @@ def _before_request():
         # let user attempt login again
         return make_basic_auth_response()
 
-    # TODO: Implement authorization
-    pass
+    # admins don't need to be authorized
+    if sender_is_admin():
+        _logger.debug(f"Admin (username={username}) authorization not required")
+        return
+
+    # authorization
+    validator = BEFORE_REQUEST_VALIDATORS.get((request.path, request.method))
+    if validator:
+        _logger.debug(f"Calling validator: {validator.__name__}")
+        if not validator():
+            return make_forbidden_response()
+    else:
+        _logger.debug(f"No validator found for {(request.path, request.method)}")
 
 
 def _after_request(resp):
     # TODO: Implement post-request logic
     return resp
```

Comment:
Can we update `after_request` to assign the MANAGE permission to the creator? For example, if I create an experiment, I should be able to update, delete, or modify permissions.

Reply:
sure, added!